src/project/internal/notationproject.cpp

```diff
@@ -779,6 +795,43 @@ Ret NotationProject::saveSelectionOnScore(const muse::io::path_t& path)
     return ret;
 }
 
+Ret NotationProject::checkSavedFileForCorruption(engraving::MscIoMode ioMode, QString path, QString scoreFileName)
+{
+    if (ioMode != MscIoMode::Zip) {
+        return muse::make_ok();
+    }
+
+    // TODO: Execute this on Windows only?
+
+    MscReader::Params params;
+    params.filePath = path;
+    params.mainFileName = scoreFileName;
+    params.mode = MscIoMode::Zip;
+
+    MscReader msczReader(params);
+    Ret ret = msczReader.open();
```

Comment:
I'm wondering if there is a faster way to check if the file is corrupted... This may significantly increase the saving time of large scores

Reply:
@cbjeukendrup any thoughts?

Reply:
In any case, having some kind of file corruption check is a million times better than not having it. Maybe we should add an option to disable it. And I don't think we need this check for autosave?

Reply:
It's a trade-off between safety and efficiency. I've seen cases of corruptions where the file was a valid ZIP file, and all content was there, except the main mscx file. So, if we really want to rule out all kinds of corruption, we'll need to read the ZIP file, extract the main mscx, and check if it isn't empty. 

I wonder how much this increases saving time in practice. All we're doing is extracting a file from a ZIP container. Since the size of mscz files is quite moderate, even for big scores, I think this is affordable; I think, when opening such scores, the ZIP reading step is only a small fraction of the reading process (obv. the layout is the step that takes most time by far, but we're not doing that here of course).

Disabling it for auto-save seems sensible. You must be _really_ unlucky when MuseScore crashes _and_ your autosave file gets corrupted (and of course if only the AutoSave file is corrupted, you still have the original version, so only your latest changes are lost).

And we could make it configurable, although I can imagine this will be a difficult decision for the user...

Reply:
msczReader.readScoreFile( ) is very fast even for the very largest scores I have - it's MscLoader::readMasterScore() that's a little slow (around 5 seconds, and most of that is in Read206::readScore - yes it's from an old version!).

Reply:
Maybe readScoreFile is fast, but uncompressing mscz isn't. And it turns out this check is called 2 times in a row. What we should do instead:

1. call it only once
2. make sure it is not called on autosave